tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -1378,17 +1403,32 @@ public TransactionType getTransactionType() {
     private final long expiration;
     private final InProgressType type;
     private final LongArrayList checkpointWritePointers;
+    private final String clientId;
```

Comment:
Not sure if we can add it to equals/hashCode. The unit tests fail when trying to compare the state before persistence and after recovery and won't be the same if we include clientId in comparison. Any reason why we should definitely include it? Or should we include it and adjust the test that does the comparison?